src/main/java/com/wojciechkocik/usage/service/TimeSpentCrossMidnightServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.wojciechkocik.usage.service;
+
+import com.wojciechkocik.usage.dto.DailyUsage;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of {@link TimeSpentCrossMidnightService} using java 8 features
+ *
+ * @author Wojciech Kocik
+ * @since 24.03.2017
+ */
+@Service
+public class TimeSpentCrossMidnightServiceImpl implements TimeSpentCrossMidnightService {
+
+    @Override
+    public List<DailyUsage> divideDaysWithTimeSessionCrossedMidnight(List<DailyUsage> dailyUsagesForCourse) {
+
+        List<DailyUsage> dividedDailyUsages = new ArrayList<>();
+
+        dailyUsagesForCourse.forEach(dailyUsage -> {
+            ZonedDateTime timeDateStarted = dailyUsage.getDateTime();
+
+            long sessionTime = dailyUsage.getTime();
+
+            long epochBeforeSession = timeDateStarted.toEpochSecond();
+            long epochAfterSession = timeDateStarted.plusSeconds(sessionTime).toEpochSecond();
+            long epochDiff = epochAfterSession - epochBeforeSession;
+
+            long untilMidnight = timeDateStarted.until(getMidnightOfDate(timeDateStarted.plusDays(1)), ChronoUnit.SECONDS);
+            long restEpoch = epochDiff - untilMidnight;
+
+            if(sessionTime <= 0){
+                return;
+            }
+            if (sessionTime <= untilMidnight) {
+                dividedDailyUsages.add(dailyUsage);
+                return;
+            }
+
+            int daysSessionExceptFirstAndLastDay = (int) Duration.ofSeconds(restEpoch).toDays();
+            int wholeDaysSession = daysSessionExceptFirstAndLastDay + 2;
```

Comment:
I've changed "+2" to "+3" and the test failed with following message:

ava.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.fail(Assert.java:95)
	at com.wojciechkocik.usage.service.TimeSpentCrossMidnightServiceTest.shouldProperlyDivideCrossedMidnightSession_parametrized(TimeSpentCrossMidnightServiceTest.java:112)
	at com.wojciechkocik.usage.service.TimeSpentCrossMidnightServiceTest.shouldProperlyDivideCrossedMidnightSessionWithChangeTimeInTimezone(TimeSpentCrossMidnightServiceTest.java:78)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) 
[..]

If this mistake has been made by accidently, I would have no idea what is going on.